### DISCUSSION TITLE
Realtime updates of specific some objects of a collection

### DISCUSSION THREAD

**Discussion by sabikrahat** [pocketbase/pocketbase#2508] [Q&A]
Hello!

I'm getting a list of `trx` from the whole collection of `transactions`.  But I want to update this query in realtime. Is there any way? I'm using Pocketbase in flutter project.

getting the list of trx code:

```
_trxs = await pb
        .collection(transactions)
        .getFullList(filter: 'gl_id = "${arg.id}"')
        .then((v) {
      log.i('Employees Trxs: $v');
      return v.map((e) {
        log.wtf('Vendor: $e');
        return PktbsTrx.fromJson(e.toJson());
      }).toList();
    });
```

I want to get only the _trxs list which satisfies my `filter: 'gl_id = "${arg.id}"'` this condition.
Example:

```
_stream() {
    pb.collection(transactions).subscribe('*', (s) {
      log.i('Stream ${s.toJson()}');
      if (s.action == 'create') {
        _trxs.add(PktbsTrx.fromJson(s.record!.toJson()));
      } else if (s.action == 'update') {
        _trxs.removeWhere((e) => e.id == s.record!.id);
        _trxs.add(PktbsTrx.fromJson(s.record!.toJson()));
      } else if (s.action == 'delete') {
        _trxs.removeWhere((e) => e.id == s.record!.id);
      }
      ref.notifyListeners();
    });
  }
```

**Comment by ganigeorgiev** [comment]
At the moment the realtime subscriptions don't support the `filter` or `expand` parameters.

You can filter manually the events from inside the callback, eg. `if (s.record.getStringValue("gl_id") != arg.id) return;`.

**Comment by sabikrahat** [comment]
Will real-time subscription support the `filter `or `expand `the parameters in future updates? @ganigeorgiev

**Reply by ganigeorgiev** [reply]
Yes, but for now is planned somewhere after the internal hooks refactoring and js app hooks integration (_you can subscribe to https://github.com/pocketbase/pocketbase/issues/2514 and https://github.com/pocketbase/pocketbase/issues/1657_).
